src/dlgSourceEditorArea.cpp

```diff
@@ -20,14 +21,51 @@
 
 
 #include "dlgSourceEditorArea.h"
+#include "edbee/edbee.h"
+#include "edbee/texteditorwidget.h"
+#include "edbee/models/textdocument.h"
+#include "edbee/views/textrenderer.h"
+#include "edbee/views/texttheme.h"
+#include "edbee/models/textgrammar.h"
+#include "edbee/models/texteditorconfig.h"
 
-#include "THighlighter.h"
-
+#include <QDir> // For the homePath()
 
 dlgSourceEditorArea::dlgSourceEditorArea(QWidget* pF) : QWidget(pF)
 {
     // init generated dialog
     setupUi(this);
-    highlighter = new THighlighter(editor->document());
-    editor->setTabStopWidth(25);
+
+    // Setting up the edbee widget
+
+    edbee::Edbee* ei = edbee::Edbee::instance();
+
+    ei->autoInit();
+    ei->autoShutDownOnAppExit();
+
+    //**********
+    //
+    // As it is, the editor only needs one syntax file (Lua). We could offer a choice of themes,
+    // but it's not really necessary just yet. People can change the theme if they really want to
+    // by overwriting the Fluidvisionlet.tmTheme file with another valid textmate theme.
+
+    // So instead of having separate theme/syntax file directories, I'm lumping them into one directory:
+    // QDir::homePath() + "/.config/mudlet/edbee/"
+
+    edbee::TextGrammarManager* grammarManager = ei->grammarManager();
+    edbee::TextGrammar* grammar = grammarManager->readGrammarFile( QDir::homePath() + "/.config/mudlet/edbee/Lua.tmLanguage" );
```

Comment:
Put this into the `mudlet_alpha.qrc` please - see `mudlet-mapper.xml` for an example of how to load it once there :) that'll ensure the file is always in the application. It won't be overwritable by the users, but we should support custom themes more properly than swapping files around anyhow.